diff -Naru linux-2.4.35.3/include/linux/nfsd/const.h linux-2.4.35.3-mct/include/linux/nfsd/const.h
--- linux-2.4.35.3/include/linux/nfsd/const.h	2007-09-24 00:02:58.000000000 +0200
+++ linux-2.4.35.3-mct/include/linux/nfsd/const.h	2007-10-03 13:10:16.000000000 +0200
@@ -25,11 +25,7 @@
  * 2) is a multiple of page size 3) can be successfully kmalloc()ed
  * by each nfsd.   
  */
-#if PAGE_SIZE > (16*1024)
 #define NFSSVC_MAXBLKSIZE	(32*1024)
-#else
-#define NFSSVC_MAXBLKSIZE	(2*PAGE_SIZE)
-#endif
 
 #ifdef __KERNEL__
 
