diff -Naru buildroot/toolchain/ccache/ccache.mk buildroot-mct/toolchain/ccache/ccache.mk
--- buildroot/toolchain/ccache/ccache.mk        2005-12-02 09:15:02.000000000 +0100
+++ buildroot-mct/toolchain/ccache/ccache.mk    2006-04-22 16:01:50.000000000 +0200
@@ -4,7 +4,7 @@
 #
 #############################################################
 CCACHE_VER:=2.4
-CCACHE_SITE:=http://ccache.samba.org/ftp/ccache
+CCACHE_SITE:=http://samba.org/ftp/ccache
 CCACHE_SOURCE:=ccache-$(CCACHE_VER).tar.gz
 CCACHE_DIR1:=$(TOOL_BUILD_DIR)/ccache-$(CCACHE_VER)
 CCACHE_DIR2:=$(BUILD_DIR)/ccache-$(CCACHE_VER)
diff -Naru buildroot/toolchain/gcc/gcc-uclibc-3.x.mk buildroot-mct/toolchain/gcc/gcc-uclibc-3.x.mk
--- buildroot/toolchain/gcc/gcc-uclibc-3.x.mk   2005-12-02 09:15:07.000000000 +0100
+++ buildroot-mct/toolchain/gcc/gcc-uclibc-3.x.mk       2006-04-22 16:00:55.000000000 +0200
@@ -19,8 +19,8 @@
 
 GCC_VERSION:=$(strip $(GCC_VERSION))
 
-#GCC_SITE:=ftp://ftp.gnu.org/gnu/gcc/releases/gcc-$(GCC_VERSION)
-GCC_SITE:=http://mirrors.rcn.net/pub/sourceware/gcc/releases/gcc-$(GCC_VERSION)
+GCC_SITE:=ftp://ftp.gwdg.de/pub/misc/gcc/releases/gcc-3.3.6
+#GCC_SITE:=http://mirrors.rcn.net/pub/sourceware/gcc/releases/gcc-$(GCC_VERSION)
 
 #
 # snapshots....
diff -Naru buildroot/toolchain/kernel-headers/kernel-headers.mk buildroot-mct/toolchain/kernel-headers/kernel-headers.mk
--- buildroot/toolchain/kernel-headers/kernel-headers.mk	2005-12-02 09:15:04.000000000 +0100
+++ buildroot-mct/toolchain/kernel-headers/kernel-headers.mk	2006-04-22 16:00:32.000000000 +0200
@@ -95,13 +95,15 @@
 	mkdir -p $(DL_DIR)
 	$(WGET) -P $(DL_DIR) $(LINUX_HEADERS_SITE)/$(LINUX_HEADERS_SOURCE)
 
-$(LINUX_HEADERS_DIR)/.unpacked: $(DL_DIR)/$(LINUX_HEADERS_SOURCE)
+#$(LINUX_HEADERS_DIR)/.unpacked: $(DL_DIR)/$(LINUX_HEADERS_SOURCE)
+$(LINUX_HEADERS_DIR)/.unpacked: 
 	rm -rf $(LINUX_HEADERS_DIR)
 	mkdir -p $(TOOL_BUILD_DIR)
-	bzcat $(DL_DIR)/$(LINUX_HEADERS_SOURCE) | tar -C $(TOOL_BUILD_DIR) $(TAR_OPTIONS) -
-ifneq ($(LINUX_HEADERS_UNPACK_DIR),$(LINUX_HEADERS_DIR))
-	mv $(LINUX_HEADERS_UNPACK_DIR) $(LINUX_HEADERS_DIR)
-endif
+	# bzcat $(DL_DIR)/$(LINUX_HEADERS_SOURCE) | tar -C $(TOOL_BUILD_DIR) $(TAR_OPTIONS) -
+	ln -s ../../../../system/linux/src $(LINUX_HEADERS_DIR)
+#ifneq ($(LINUX_HEADERS_UNPACK_DIR),$(LINUX_HEADERS_DIR))
+#	mv $(LINUX_HEADERS_UNPACK_DIR) $(LINUX_HEADERS_DIR)
+#endif
 	touch $(LINUX_HEADERS_DIR)/.unpacked
 
 $(LINUX_HEADERS_DIR)/.patched: $(LINUX_HEADERS_DIR)/.unpacked
diff -Naru buildroot/toolchain/uClibc/uClibc.config buildroot-mct/toolchain/uClibc/uClibc.config
--- buildroot/toolchain/uClibc/uClibc.config	2005-12-02 09:15:02.000000000 +0100
+++ buildroot-mct/toolchain/uClibc/uClibc.config	2006-04-22 16:00:32.000000000 +0200
@@ -12,7 +12,7 @@
 # TARGET_i960 is not set
 # TARGET_m68k is not set
 # TARGET_microblaze is not set
-# TARGET_mips is not set
+TARGET_mips=y
 # TARGET_nios is not set
 # TARGET_nios2 is not set
 # TARGET_powerpc is not set
@@ -26,8 +26,13 @@
 # Target Architecture Features and Options
 #
 HAVE_ELF=y
-TARGET_ARCH="none"
-# ARCH_HAS_NO_MMU is not set
+ARCH_SUPPORTS_LITTLE_ENDIAN=y
+TARGET_ARCH="mips"
+ARCH_CFLAGS="-mno-split-addresses"
+ARCH_SUPPORTS_BIG_ENDIAN=y
+CONFIG_MIPS_ISA_1=y
+ARCH_LITTLE_ENDIAN=y
+ARCH_HAS_NO_MMU=y
 UCLIBC_HAS_FLOATS=y
 HAS_FPU=y
 DO_C99_MATH=y
@@ -49,7 +54,7 @@
 LDSO_CACHE_SUPPORT=y
 # LDSO_PRELOAD_FILE_SUPPORT is not set
 LDSO_BASE_FILENAME="ld.so"
-LDSO_RUNPATH=n
+LDSO_RUNPATH=y
 # DL_FINI_CRT_COMPAT is not set
 UCLIBC_CTOR_DTOR=y
 # HAS_NO_THREADS is not set
@@ -107,7 +112,7 @@
 UCLIBC_HAS_STDIO_BUILTIN_BUFFER_NONE=y
 # UCLIBC_HAS_STDIO_BUILTIN_BUFFER_4 is not set
 # UCLIBC_HAS_STDIO_BUILTIN_BUFFER_8 is not set
-UCLIBC_HAS_STDIO_SHUTDOWN_ON_ABORT=n
+UCLIBC_HAS_STDIO_SHUTDOWN_ON_ABORT=y
 UCLIBC_HAS_STDIO_GETC_MACRO=y
 UCLIBC_HAS_STDIO_PUTC_MACRO=y
 UCLIBC_HAS_STDIO_AUTO_RW_TRANSITION=y
@@ -116,9 +121,9 @@
 UCLIBC_HAS_GLIBC_CUSTOM_STREAMS=y
 UCLIBC_HAS_PRINTF_M_SPEC=y
 UCLIBC_HAS_ERRNO_MESSAGES=y
-# UCLIBC_HAS_SYS_ERRLIST is not set
+UCLIBC_HAS_SYS_ERRLIST=y
 UCLIBC_HAS_SIGNUM_MESSAGES=y
-# UCLIBC_HAS_SYS_SIGLIST is not set
+UCLIBC_HAS_SYS_SIGLIST=y
 UCLIBC_HAS_GNU_GETOPT=y
 
 #
diff -Naru buildroot/toolchain/uClibc/uClibc.config-locale buildroot-mct/toolchain/uClibc/uClibc.config-locale
--- buildroot/toolchain/uClibc/uClibc.config-locale	2005-12-02 09:15:02.000000000 +0100
+++ buildroot-mct/toolchain/uClibc/uClibc.config-locale	2006-04-22 16:00:32.000000000 +0200
@@ -12,7 +12,7 @@
 # TARGET_i960 is not set
 # TARGET_m68k is not set
 # TARGET_microblaze is not set
-# TARGET_mips is not set
+TARGET_mips=y
 # TARGET_nios is not set
 # TARGET_nios2 is not set
 # TARGET_powerpc is not set
@@ -26,13 +26,25 @@
 # Target Architecture Features and Options
 #
 HAVE_ELF=y
-TARGET_ARCH="none"
+ARCH_SUPPORTS_LITTLE_ENDIAN=y
+TARGET_ARCH="mips"
+ARCH_CFLAGS="-mno-split-addresses"
+ARCH_SUPPORTS_BIG_ENDIAN=y
+CONFIG_MIPS_ISA_1=y
+# CONFIG_MIPS_ISA_2 is not set
+# CONFIG_MIPS_ISA_3 is not set
+# CONFIG_MIPS_ISA_4 is not set
+# CONFIG_MIPS_ISA_MIPS32 is not set
+# CONFIG_MIPS_ISA_MIPS64 is not set
+ARCH_LITTLE_ENDIAN=y
+# ARCH_BIG_ENDIAN is not set
 # ARCH_HAS_NO_MMU is not set
+ARCH_HAS_MMU=y
 UCLIBC_HAS_FLOATS=y
 HAS_FPU=y
 DO_C99_MATH=y
-WARNINGS="-Wall"
 KERNEL_SOURCE="/usr/src/linux"
+C_SYMBOL_PREFIX=""
 HAVE_DOT_CONFIG=y
 
 #
@@ -49,12 +61,12 @@
 LDSO_CACHE_SUPPORT=y
 # LDSO_PRELOAD_FILE_SUPPORT is not set
 LDSO_BASE_FILENAME="ld.so"
+LDSO_RUNPATH=y
 # DL_FINI_CRT_COMPAT is not set
 UCLIBC_CTOR_DTOR=y
 # HAS_NO_THREADS is not set
 UCLIBC_HAS_THREADS=y
 PTHREADS_DEBUG_SUPPORT=y
-LINUXTHREADS_OLD=y
 UCLIBC_HAS_LFS=y
 # UCLIBC_STATIC_LDCONFIG is not set
 # MALLOC is not set
@@ -62,7 +74,6 @@
 MALLOC_STANDARD=y
 MALLOC_GLIBC_COMPAT=y
 UCLIBC_DYNAMIC_ATEXIT=y
-# COMPAT_ATEXIT is not set
 HAS_SHADOW=y
 # UNIX98PTY_ONLY is not set
 ASSUME_DEVPTS=y
@@ -111,6 +122,7 @@
 UCLIBC_HAS_STDIO_BUILTIN_BUFFER_NONE=y
 # UCLIBC_HAS_STDIO_BUILTIN_BUFFER_4 is not set
 # UCLIBC_HAS_STDIO_BUILTIN_BUFFER_8 is not set
+UCLIBC_HAS_STDIO_SHUTDOWN_ON_ABORT=y
 UCLIBC_HAS_STDIO_GETC_MACRO=y
 UCLIBC_HAS_STDIO_PUTC_MACRO=y
 UCLIBC_HAS_STDIO_AUTO_RW_TRANSITION=y
@@ -119,9 +131,9 @@
 UCLIBC_HAS_GLIBC_CUSTOM_STREAMS=y
 UCLIBC_HAS_PRINTF_M_SPEC=y
 UCLIBC_HAS_ERRNO_MESSAGES=y
-# UCLIBC_HAS_SYS_ERRLIST is not set
+UCLIBC_HAS_SYS_ERRLIST=y
 UCLIBC_HAS_SIGNUM_MESSAGES=y
-# UCLIBC_HAS_SYS_SIGLIST is not set
+UCLIBC_HAS_SYS_SIGLIST=y
 UCLIBC_HAS_GNU_GETOPT=y
 
 #
@@ -149,6 +161,7 @@
 #
 CROSS_COMPILER_PREFIX=""
 # DODEBUG is not set
+# DODEBUG_PT is not set
 # DOASSERTS is not set
 # SUPPORT_LD_DEBUG is not set
 # SUPPORT_LD_DEBUG_EARLY is not set
diff -Naru buildroot/toolchain/uClibc/uclibc.mk buildroot-mct/toolchain/uClibc/uclibc.mk
--- buildroot/toolchain/uClibc/uclibc.mk	2005-12-02 09:15:02.000000000 +0100
+++ buildroot-mct/toolchain/uClibc/uclibc.mk	2006-04-22 16:01:15.000000000 +0200
@@ -15,9 +15,10 @@
 ifeq ($(BR2_UCLIBC_VERSION_SNAPSHOT),y)
 # Be aware that this changes daily....
 UCLIBC_DIR:=$(TOOL_BUILD_DIR)/uClibc
-UCLIBC_SOURCE:=uClibc-$(strip $(subst ",, $(BR2_USE_UCLIBC_SNAPSHOT))).tar.bz2
+#UCLIBC_SOURCE:=uClibc-$(strip $(subst ",, $(BR2_USE_UCLIBC_SNAPSHOT))).tar.bz2
 #"
-UCLIBC_SITE:=http://www.uclibc.org/downloads/snapshots
+UCLIBC_SOURCE:=uClibc-20050828.tar.bz2
+UCLIBC_SITE:=http://www.openmct.org/pub/build
 else
 UCLIBC_VER:=0.9.28
 UCLIBC_DIR:=$(TOOL_BUILD_DIR)/uClibc-$(UCLIBC_VER)
