diff -Naru samba-3.0.23c/source/configure samba-3.0.23c-mct/source/configure
--- samba-3.0.23c/source/configure	2006-08-31 20:57:32.000000000 +0200
+++ samba-3.0.23c-mct/source/configure	2006-09-20 23:21:00.000000000 +0200
@@ -14438,7 +14438,7 @@
 fi
 echo "$as_me:$LINENO: result: $samba_cv_have_setresuid" >&5
 echo "${ECHO_T}$samba_cv_have_setresuid" >&6
-if test x"$samba_cv_have_setresuid" = x"yes"; then
+if test x"$samba_cv_have_setresuid" = x"yes" || test x"$samba_cv_have_setresuid" = x"cross"; then
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_SETRESUID 1
@@ -14492,7 +14492,7 @@
 fi
 echo "$as_me:$LINENO: result: $samba_cv_have_setresgid" >&5
 echo "${ECHO_T}$samba_cv_have_setresgid" >&6
-if test x"$samba_cv_have_setresgid" = x"yes"; then
+if test x"$samba_cv_have_setresgid" = x"yes" || test x"$samba_cv_have_setresgid" = x"cross"; then
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_SETRESGID 1
@@ -21195,11 +21195,7 @@
     *linux*)
        # glibc <= 2.3.2 has a broken getgrouplist
        if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+          linux_getgrouplist_ok=no
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -27818,7 +27814,7 @@
 fi
 echo "$as_me:$LINENO: result: $samba_cv_HAVE_GETTIMEOFDAY_TZ" >&5
 echo "${ECHO_T}$samba_cv_HAVE_GETTIMEOFDAY_TZ" >&6
-if test x"$samba_cv_HAVE_GETTIMEOFDAY_TZ" = x"yes"; then
+if test x"$samba_cv_HAVE_GETTIMEOFDAY_TZ" = x"yes" || test x"$samba_cv_HAVE_GETTIMEOFDAY_TZ" = x"cross"; then
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_GETTIMEOFDAY_TZ 1
