diff -Naru portmap_4/Makefile portmap_4-mct/Makefile
--- portmap_4/Makefile	1996-05-31 15:50:40.000000000 +0200
+++ portmap_4-mct/Makefile	2005-11-03 13:06:03.608369381 +0100
@@ -46,7 +46,7 @@
 # libwrap.a object library. WRAP_DIR should specify the directory with
 # that library.
 
-WRAP_DIR= ../tcp_wrappers
+WRAP_DIR= ../tcp_wrappers_7.6
 
 # Auxiliary object files that may be missing from your C library.
 #
@@ -69,7 +69,7 @@
 
 COPT	= -Dconst= -Dperror=xperror $(HOSTS_ACCESS) $(CHECK_PORT) \
 	$(SYS) -DFACILITY=$(FACILITY) $(ULONG) $(ZOMBIES) $(SA_LEN)
-CFLAGS	= $(COPT) -O $(NSARCHS) $(SETPGRP)
+CFLAGS	= $(COPT) -Os $(NSARCHS) $(SETPGRP)
 OBJECTS	= portmap.o pmap_check.o from_local.o $(AUX)
 
 all:	portmap pmap_dump pmap_set
diff -Naru portmap_4/portmap.c portmap_4-mct/portmap.c
--- portmap_4/portmap.c	1996-05-31 15:52:59.000000000 +0200
+++ portmap_4-mct/portmap.c	2005-11-03 13:05:34.189873791 +0100
@@ -235,7 +235,7 @@
 /* need to override perror calls in rpc library */
 void
 perror(what)
-	const char *what;
+	__const char *what;
 {
 
 	syslog(LOG_ERR, "%s: %m", what);
