diff -Naru linux-2.4.35.3/arch/mips/kernel/syscalls.h linux-2.4.35.3-mct/arch/mips/kernel/syscalls.h
--- linux-2.4.35.3/arch/mips/kernel/syscalls.h	2007-09-24 00:02:58.000000000 +0200
+++ linux-2.4.35.3-mct/arch/mips/kernel/syscalls.h	2007-10-03 13:07:54.000000000 +0200
@@ -255,3 +255,36 @@
 SYS(sys_ni_syscall, 0)				/* res. for futex */
 SYS(sys_ni_syscall, 0)				/* res. for sched_setaffinity */
 SYS(sys_ni_syscall, 0)				/* 4240 res. for sched_getaffinity */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4245 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4250 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4255 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4260 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4265 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)				/* 4270 */
+SYS(sys_ni_syscall, 0)
+SYS(sys_ni_syscall, 0)
+SYS(sys_vserver, 0)				/* 4273 vserver */
diff -Naru linux-2.4.35.3/include/asm-mips/unistd.h linux-2.4.35.3-mct/include/asm-mips/unistd.h
--- linux-2.4.35.3/include/asm-mips/unistd.h	2007-09-24 00:02:58.000000000 +0200
+++ linux-2.4.35.3-mct/include/asm-mips/unistd.h	2007-10-03 13:07:54.000000000 +0200
@@ -257,16 +257,17 @@
 #define __NR_futex			(__NR_Linux + 238)
 #define __NR_sched_setaffinity		(__NR_Linux + 239)
 #define __NR_sched_getaffinity		(__NR_Linux + 240)
+#define __NR_vserver			(__NR_Linux + 273)
 
 /*
  * Offset of the last Linux o32 flavoured syscall
  */
-#define __NR_Linux_syscalls		240
+#define __NR_Linux_syscalls		273
 
 #endif /* _MIPS_SIM == _MIPS_SIM_ABI32 */
 
 #define __NR_O32_Linux			4000
-#define __NR_O32_Linux_syscalls		240
+#define __NR_O32_Linux_syscalls		273
 
 #if _MIPS_SIM == _MIPS_SIM_ABI64
 
