--- src/include/usage.h.orig	2007-11-04 07:03:24.000000000 +0100
+++ src/include/usage.h	2010-08-13 21:51:24.183156283 +0200
@@ -3833,7 +3833,7 @@
        "\n	-C,--clientid-none	Suppress default client identifier" \
        "\n	-p,--pidfile=file	Store process ID of daemon in file" \
        "\n	-r,--request=IP		IP address to request" \
-       "\n	-s,--script=file	Run file at dhcp events (default: /usr/share/udhcpc/default.script)" \
+       "\n	-s,--script=file	Run file at dhcp events (default: /sbin/default.script)" \
        "\n	-t,--retries=N		Send up to N request packets"\
        "\n	-f,--foreground	Run in foreground" \
        "\n	-b,--background	Background if lease cannot be immediately negotiated" \
@@ -3851,7 +3851,7 @@
        "\n	-C		Suppress default client identifier" \
        "\n	-p file		Store process ID of daemon in file" \
        "\n	-r IP		IP address to request" \
-       "\n	-s file		Run file at dhcp events (default: /usr/share/udhcpc/default.script)" \
+       "\n	-s file		Run file at dhcp events (default: /sbin/default.script)" \
        "\n	-t N		Send up to N request packets"\
        "\n	-f		Run in foreground" \
        "\n	-b		Background if lease cannot be immediately negotiated" \
